### PR TITLE
Test retain flag

### PR DESCRIPTION
**Pull request by graebm**
Split single test into 2 to ensure we're sending publish flags correctly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/packet_encoding_test.c [hunk 1]
```diff
@@ -288,22 +288,61 @@ PACKET_TEST(CONNACK, connack, &s_test_connack_init, NULL, NULL)
...
+        aws_byte_cursor_from_array(s_payload, PAYLOAD_LEN)));
+
+    /* Init buffer */
+    /* clang-format off */
+    aws_byte_buf_write_u8(
+        &fixture->buffer,
+        (AWS_MQTT_PACKET_PUBLISH << 4) /* Packet type bits 7-4 */
+        | (1 << 3) /* DUP bit 3 */
+        | (AWS_MQTT_QOS_AT_MOST_ONCE << 1) /* QoS bits 2-1 */
+        | (0 << 0) /* RETAIN bit 0 */);
```
Comment (justinboswell) [inline tests/packet_encoding_test.c:310]: Shouldn't this be 1 << 0?

Comment (TingDaoK) [reply tests/packet_encoding_test.c:310]: retain is set to false in this test, so it's correct to be zero here.

**Review by TingDaoK — APPROVED** [review]
LGTM

File: tests/packet_encoding_test.c (hunk 1 above)
Comment (graebm) [reply tests/packet_encoding_test.c:310]: it's correct, but yeah, bonkers looking. I'll remove the useless shift

**Pull request merged**